src/allocators/allocator.rs

```diff
@@ -242,84 +242,4 @@ pub trait Allocator: Debug + Sized {
     ) -> Result<MemoryAddress, AllocErr> {
         self.reallocate(ptr, layout, new_size)
     }
-
-    #[doc(hidden)]
-    #[inline(always)]
-    unsafe fn alloc_alloc_excess(&self, layout: Layout) -> Result<Excess, AllocErr> {
-        let layout = LayoutHack::access_private_fields(layout);
-        if unlikely!(layout.size_ == 0) {
-            return Ok(Excess(Self::ZERO_SIZED_ALLOCATION, 0));
-        }
```

Comment:
Where did these go? Do new traits inherit this behavior?

Reply:
As far as I understanding, there's no more method called as the `alloc_alloc_excess`. However, looks like need to use here `https://doc.rust-lang.org/1.30.1/alloc/alloc/trait.Alloc.html#method.alloc_excess` instead?

Reply:
good catch, i think i forgot to put them back

Reply:
For example shrink and grow are now provided methods https://doc.rust-lang.org/std/alloc/trait.AllocRef.html#provided-methods

I'm not too aware of the changes but it seems [like it's not needed anymore](https://github.com/rust-lang/rust/issues/32838#issuecomment-208141759)

grow and shrink seem to be enough, and they are _provided_ by [AllocRef](https://doc.rust-lang.org/std/alloc/trait.AllocRef.html#provided-methods)

Reply:
* `alloc_alloc_excess` is our method here maps to alloc_excess in all allocators.
* alloc_zeroed should be implemented because https://doc.rust-lang.org/std/io/struct.Initializer.html uses it behind the scenes.

Reply:
But one can send non zeroed for zeroed allocation, might have better perf impact, but not does what it supposed to do.